vllm/transformers_utils/tokenizer.py

```diff
@@ -99,45 +102,64 @@ def get_tokenizer(
         kwargs["gguf_file"] = Path(tokenizer_name).name
         tokenizer_name = Path(tokenizer_name).parent
 
-    try:
-        tokenizer = AutoTokenizer.from_pretrained(
-            tokenizer_name,
-            *args,
-            trust_remote_code=trust_remote_code,
-            revision=revision,
-            **kwargs)
-    except ValueError as e:
-        # If the error pertains to the tokenizer class not existing or not
-        # currently being imported, suggest using the --trust-remote-code flag.
-        if (not trust_remote_code and
-            ("does not exist or is not currently imported." in str(e)
-             or "requires you to execute the tokenizer file" in str(e))):
-            err_msg = (
-                "Failed to load the tokenizer. If the tokenizer is a custom "
-                "tokenizer not yet available in the HuggingFace transformers "
-                "library, consider setting `trust_remote_code=True` in LLM "
-                "or using the `--trust-remote-code` flag in the CLI.")
-            raise RuntimeError(err_msg) from e
-        else:
-            raise e
-    except AttributeError as e:
-        if "BaichuanTokenizer" in str(e):
-            # This is for the error "'BaichuanTokenizer' object has no
-            # attribute 'sp_model'".
-            tokenizer = BaichuanTokenizer.from_pretrained(
+    # if tokenizer is from official mistral org
+    is_from_mistral_org = str(tokenizer_name).split("/")[0] == "mistralai"
+    if is_from_mistral_org and tokenizer_mode != "mistral":
+        warnings.warn(
```

Comment:
For now, I've just added a FutureWarning when people load mistral models **without** mistral tokenizers. 
In the future, I'd love to make `tokenizer_mode == "mistral"` the default for Mistral models, e.g. so that this works out of the box:

```py
outputs = llm.chat(messages)
```

Which currently leads to:
```
ValueError: As of transformers v4.44, default chat template is no longer allowed, so you must provide a chat template if the tokenizer does not define one.
```

Reply:
Matt from HF here - you're making me paranoid! All of the Mistral `Instruct` models in `mistralai/` should have chat templates, so that error should only appear when you use a base model that hasn't been trained for chat, right?

Reply:
```py
from vllm import LLM, SamplingParams

sampling_params = SamplingParams(temperature=0.8, top_p=0.95)

for model in ["mistralai/Mistral-Nemo-Base-2407"]: # or "mistralai/Mistral-7B-Instruct-v0.3"
    llm = LLM(model=model, max_model_len=8192)

    prompt1 = "Explain the concept of entropy."
    messages = [
        {
            "role": "system",
            "content": "You are a helpful assistant"
        },
        {
            "role": "user",
            "content": prompt1
        },
    ]
    outputs = llm.chat(messages)

    print(outputs)
```

doesn't seem to work. Only if you add `tokenizer_mode="mistral"`

Reply:
Generally, we'd be happy to have to rely less on the Jinja templates for Mistral models and more on the well-tested mistral_common for tokenization (if possible)

Reply:
Yes! The problem here occurs because we generally only have chat templates on chat models. Base models like `Nemo-Base` haven't been trained for chat. In general, base models on HF don't have chat templates so users don't get low-quality output when they input chat tokens into a model that wasn't trained for them.

However, if you prefer, we can totally add chat templates to `Nemo-Base` and the other base models. The HF tokenizer should give identical outputs to the tokenizers in `mistral-common` now, and the templating should give identical results too - I tested this for a big range of models, so if we add templates to the base models that should get them to 1:1 equivalence as well.

That said, if you prefer to use `mistral-common` rather than Jinja templates, that's totally understandable, lol. I just wanted to make clear the Base model thing is a simple fix, and I'm happy to help with it!